_posts/contests/2021-02-05-Performance-analysis-and-tuning-contest-5.md

```diff
@@ -0,0 +1,216 @@
+---
+layout: post
+title: Performance analysis and tuning challenge &#35&#53.
+categories: challenge
+author: Ivica Bogosavljevic from johnysswlab.com
+---
+
+**Contents:**
+* TOC
+{:toc}
+
+Welcome to the 5th edition of our performance analysis and tuning challenge. If you haven't participated in our challenges before, we highly encourage you to read the [introductory post]({{ site.url }}/blog/2019/02/02/Performance-optimization-contest) first. 
+
+The fifth edition of the contest will be run by Ivica Bogosavljevic from [Johny's Software Lab](https://johnysswlab.com) blog. Ivica also writes about software performance, so feel free to go and check out his blog, there is a ton of useful content there.
+
+The benchmark for the 5th edition is KALDI. Kaldi is an open-source toolkit for speech recognition written in C++ intended to be used by researchers.
+
+In the essence, Kaldi takes an input model and recorded speech and then it converts it to a textual representation. Speech recognition is a complex topic, but as always, we focus on hardware efficiency and general data processing efficiency, not on the speech recognition algorithm itself.
+
+## Quickstart
+
+Here are instructions on how to build Kaldi on Linux. To download and build kaldi for the first time do the following:
+
+```bash
+$ git clone https://github.com/kaldi-asr/kaldi.git
+$ cd kaldi
+$ git checkout ca6d133262aa183b23f6daba48995bd7576fb572
```

Comment:
Do we need some particular commit?
If we do, then it would be good to explain why it is so.

Reply:
I checked against it and I know it works. I tried with some earlier commit, it didn't work because some options were not supported back then. It's better for everyone to simply use the same commit

Reply:
Ah, I see, you want to fix the baseline code that way.
How about we fork the kaldi repo? This will do the trick.
Also, I'm thinking we can add a CI runner to the forked repo. Will send more details in the email.